### PR TITLE
use dash instead of underscore

### PR DESCRIPTION
**Pull request by myleshorton**
This is because we don't use underscores anywhere else. the i18n package is able to figure out the translation file to use either way (the translation files are using "-" in their name), but other things are not necessarily equipped to use "_", such as the pro server.

### CODE REVIEW

File: translate.go [hunk 1]
```diff
@@ -107,8 +107,9 @@ func SetLocale(locale string) (string, error) {
 	if matched, _ := regexp.MatchString(localeRegexp, locale); !matched {
 		return "", fmt.Errorf("Malformated locale string %s", locale)
 	}
-	locale = strings.Replace(locale, "-", "_", -1)
-	parts := strings.Split(locale, "_")
+	const sep = "-"
```
Comment (oxtoacart) [inline translate.go:110]: Might as well make this a global constant.

**Comment by oxtoacart** [comment]
Test fails for me with:

```
DEBUG i18n: translate.go:114 Setting locale en-US
DEBUG i18n: translate.go:114 Setting locale en-US
DEBUG i18n: translate.go:114 Setting locale en
DEBUG i18n: translate.go:114 Setting locale en-US
DEBUG i18n: translate.go:114 Setting locale zh-CN
--- FAIL: TestTranslate (0.01s)
	translate_test.go:91: Expect T("ONLY_IN_ZH_CN") to be "I speak Mandarin!", got "[ONLY_IN_ZH_CN]"
DEBUG i18n: translate.go:114 Setting locale en-US
DEBUG i18n: translate.go:114 Setting locale zh-CN
--- FAIL: TestReadFromMemory (0.00s)
	translate_test.go:91: Expect T("ONLY_IN_ZH") to be "I speak Chinese!", got "[ONLY_IN_ZH]"
DEBUG i18n: translate.go:114 Setting locale en-US
FAIL
exit status 1
FAIL	github.com/getlantern/i18n	0.050s
```

**Comment by myleshorton** [comment]
OK good catch on tests. Just fixed.

File: translate.go (hunk 1 above)
Comment (myleshorton) [reply translate.go:110]: To me this is clearer TBH. It's the only place it's used, and if it's global the reader would have to then search for where it's used. This way is more obvious.

Comment (oxtoacart) [reply translate.go:110]: I'll buy that.

**Pull request merged**